firmware/application/apps/ble_app.hpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2014 Jared Boone, ShareBrained Technology, Inc.
+ * Copyright (C) 2017 Furrtek
+ * Copyright (C) 2023 TJ Baginski
+ *
+ * This file is part of PortaPack.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifndef __BLE_APP_H__
+#define __BLE_APP_H__
+
+#include "ui.hpp"
+#include "ui_navigation.hpp"
+#include "ui_receiver.hpp"
+#include "ui_freq_field.hpp"
+#include "ui_record_view.hpp"
+#include "app_settings.hpp"
+#include "radio_state.hpp"
+#include "log_file.hpp"
+#include "utility.hpp"
+
+#include "recent_entries.hpp"
+
+class BLELogger {
+   public:
+    Optional<File::Error> append(const std::string& filename) {
+        return log_file.append(filename);
+    }
+
+    void log_raw_data(const std::string& data);
+
+   private:
+    LogFile log_file{};
+};
+
+namespace ui {
+typedef enum {
```

Comment:
just a general nit: I'm kind of torn on this advice myself.
C++ has `enum class` which provides for scoped enumeration values.
On one hand they are nice because they don't pollute the namespace, not the other hand they are very verbose (at least until later compilers added support for `using`.

I'm just pointing it out, not that you need to make a change here, but C++ leans toward using them.

Another nice thing is that you can specify the size if you want to control how many bytes are used.

Also, in C++ don't `typedef` things like structs and enums, just declare them like `enum class Foobar {` -- generally don't `typedef` and prefer `using` which has nicer aliasing syntax anyway.

Reply:
Done Changed it to fit C++.